slang_frontend.cc

```diff
@@ -985,7 +986,14 @@ struct ProceduralVisitor : public ast::ASTVisitor<ProceduralVisitor, true, false
 			return;
 		}
 
+		RTLIL::Wire *disable = netlist.canvas->addWire(NEW_ID_SUFFIX("disable"), 1);
+		disable->attributes[ID($nonstatic)] = current_case->level;
+		do_simple_assign(stmt.sourceRange.start(), disable, RTLIL::S0, true);
+
 		while (true) {
+			SwitchHelper b(current_case, vstate, {RTLIL::S0});
+			b.sw->statement = &stmt;
```

Comment:
Please move the switch helper construction and the `b.finish(netlist)` call closer to the branch call. This will create a superfluous switch at the end when we are exiting out of the loop.

Reply:
Done!